### PR TITLE
Memory metering

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by wanderer** [comment]
lgtm

File: metering.md [hunk 1]
```diff
@@ -20,10 +20,31 @@ Metering is done by counting the cost of running a continuous subtree of the AST
...
+
+Memory size is expressed in pages, where a page is 65536 bytes.
+
+The module parameter specifies the initial page count and an optional maximal page count the module cannot exceed. The currently available page count can be queried via `current_memory` and new pages can be allocated via `grow_memory`. Read more about memory in the [the WebAssembly design](https://github.com/WebAssembly/design/blob/master/Modules.md#linear-memory-section).
+
+Gas needs to be charged for the initial allocated pages as well as any increase of pages via `grow_memory`.
+
+### Initial memory allocation
+
+Metering call needs to be injected as the very first instruction if preallocated memory pages are defined.
```
Comment (wanderer) [inline metering.md]: Metering Calls are always injected as the very first instruction regardless if the is preallocated memory or not

Comment (axic) [inline metering.md]: The pre-allocated memory page is defined by `memory` in the module section as opposed to opcodes in the `func` section. So it is not part of what the regular metering code does.

Comment (wanderer) [inline metering.md]: yep, but i think i'm confused on this part

> Metering call needs to be injected as the very first instruction if preallocated memory pages are defined.

Metering costs need to added to the first metering call. The first instuction to run is always a metering call.

Comment (axic) [inline metering.md]: Perhaps this is better wording:

> The cost of pre-allocated memory must be included in the very first metering call.

**Pull request merged**